tests/src/utils/utils.ts

```diff
@@ -0,0 +1,35 @@
+import { KeyringPair } from '@polkadot/keyring/types';
+import { SubmittableExtrinsic } from '@polkadot/api/types';
+import BN = require('bn.js');
+
+export class Utils {
+  public static async signAndSend(
+    tx: SubmittableExtrinsic<'promise'>,
+    account: KeyringPair,
+    nonce: BN,
+    expectFailure = false
+  ): Promise<void> {
+    return new Promise(async (resolve, reject) => {
+      const signedTx = tx.sign(account, { nonce });
+
+      await signedTx
+        .send(async result => {
+          if (result.status.isFinalized === true && result.events !== undefined) {
+            result.events.forEach(event => {
+              if (event.event.method === 'ExtrinsicFailed') {
+                if (expectFailure) {
```

Comment:
I think we discussed this on a call, this is fine for now, but in future to be able to do more elaborate assertions on result of the extrinsic, the promise will need to always resolve with a "Result" type that can be inspected to determine the outcome of the extrinsic.